internal/provider/base.go

```diff
@@ -30,4 +41,19 @@ func (p *DefaultProvider) Name() models.PlatformType {
 // AddPlatformToInstallConfig adds the current platform to the installconfig.
 func (p *DefaultProvider) AddPlatformToInstallConfig(_ *installcfg.InstallerConfigBaremetal, _ *common.Cluster) error {
 	return nil
-}
\ No newline at end of file
+}
```

Comment:
Let's remove the `DefaultProvider` to avoid this error-prone behavior that would occur only in runtime.
I'd prefer to have explicitly every provider implementation (even if it's empty) and to fail on compilation time when something is missing.

Reply:
Done